### ISSUE TITLE
Refactor ignored files

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#958]
Right now, the ignored files reading is done in `config.js`. It really doesn't belong there as it has nothing to do with the configuration that is passed around.

We should extract all ignore file functionality into a new utility, such that we can do:

``` js
var exclusions = Exclusions.load(options);

if (!exclusions.contains(filePath)) {
    // process the file or directory
}
```
